code/modules/mob_spawn/mob_spawn.dm

```diff
@@ -164,6 +175,7 @@
 	if(QDELETED(src) || QDELETED(user))
 		return
 	user.log_message("became a [prompt_name].", LOG_GAME)
+	currently_deciding_ghosts -= user.key
```

Comment:
If this early returns, keys won't be properly removed from currently_deciding_ghosts. You should probably do this a lot earlier, like as soon as the prompt has ended.

Reply:
Good catch